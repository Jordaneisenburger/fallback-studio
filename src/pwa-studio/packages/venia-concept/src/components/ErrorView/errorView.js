import React, { Component } from 'react';

import { fullPageLoadingIndicator } from '../LoadingIndicator';

const messages = new Map()
    .set('loading', fullPageLoadingIndicator)
    .set('notFound', 'That page could not be found. Please try again.')
    .set('internalError', 'Something went wrong. Please try again.')
    .set(
        'outOfStock',
        'This Product is currently out of stock. Please try again later.'
    );

class ErrorView extends Component {
    render() {
        const { loading, notFound, outOfStock } = this.props;
        const message = loading
            ? messages.get('loading')
            : notFound
            ? messages.get('notFound')
            : outOfStock
            ? messages.get('outOfStock')
            : messages.get('internalError');

        return <h1>{message}</h1>;
    }
}

export default ErrorView;
