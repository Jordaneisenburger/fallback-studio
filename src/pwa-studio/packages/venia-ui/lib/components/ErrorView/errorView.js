import React, { useMemo } from 'react';

import { fullPageLoadingIndicator } from '../LoadingIndicator';

const messages = new Map()
    .set('loading', fullPageLoadingIndicator)
    .set('notFound', 'That page could not be found. Please try again.')
    .set('internalError', 'Something went wrong. Please try again.')
    .set(
        'outOfStock',
        'This Product is currently out of stock. Please try again later.'
    );

const ErrorView = props => {
    const { loading, notFound, outOfStock } = props;

    const message = useMemo(() => {
        return loading
            ? messages.get('loading')
            : notFound
            ? messages.get('notFound')
            : outOfStock
            ? messages.get('outOfStock')
            : messages.get('internalError');
    }, [loading, notFound, outOfStock]);

    return <h1>{message}</h1>;
};

export default ErrorView;
