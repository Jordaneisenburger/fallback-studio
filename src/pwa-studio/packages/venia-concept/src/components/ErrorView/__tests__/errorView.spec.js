import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import ErrorView from '../errorView';

const renderer = new ShallowRenderer();

test('it renders the correct tree when loading', () => {
    const tree = renderer.render(<ErrorView loading={true} />);

    expect(tree).toMatchSnapshot();
});

test('it renders the correct tree when page not found', () => {
    const tree = renderer.render(<ErrorView notFound={true} />);

    expect(tree).toMatchSnapshot();
});

test('it renders the correct tree when out of stock', () => {
    const tree = renderer.render(<ErrorView outOfStock={true} />);

    expect(tree).toMatchSnapshot();
});

test('it renders the internal error tree otherwise', () => {
    const tree = renderer.render(<ErrorView />);

    expect(tree).toMatchSnapshot();
});
