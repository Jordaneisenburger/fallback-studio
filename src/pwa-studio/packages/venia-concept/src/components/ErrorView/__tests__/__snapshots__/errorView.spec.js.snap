// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders the correct tree when loading 1`] = `
<h1>
  <LoadingIndicator
    global={true}
  >
    Fetching Data...
  </LoadingIndicator>
</h1>
`;

exports[`it renders the correct tree when out of stock 1`] = `
<h1>
  This Product is currently out of stock. Please try again later.
</h1>
`;

exports[`it renders the correct tree when page not found 1`] = `
<h1>
  That page could not be found. Please try again.
</h1>
`;

exports[`it renders the internal error tree otherwise 1`] = `
<h1>
  Something went wrong. Please try again.
</h1>
`;
